### ISSUE TITLE
apache_request_headers() not exists when running with nginx php-fpm

### ISSUE THREAD

**Issue by zergtant** [kiswa/TaskBoard#317]
modify api/mailFactory.php 
add follow code 

```
if( !function_exists('apache_request_headers') ) {
///
function apache_request_headers() {
  $arh = array();
  $rx_http = '/\AHTTP_/';
  foreach($_SERVER as $key => $val) {
    if( preg_match($rx_http, $key) ) {
      $arh_key = preg_replace($rx_http, '', $key);
      $rx_matches = array();
      // do some nasty string manipulations to restore the original letter case
      // this should work in most cases
      $rx_matches = explode('_', $arh_key);
      if( count($rx_matches) > 0 and strlen($arh_key) > 2 ) {
        foreach($rx_matches as $ak_key => $ak_val) $rx_matches[$ak_key] = ucfirst($ak_val);
        $arh_key = implode('-', $rx_matches);
      }
      $arh[$arh_key] = $val;
    }
  }
  return( $arh );
}
///
}
```

before function getServerHost()
will solve this problem

**Comment by kiswa** [comment]
I'll have to look into this to make sure it's not a problem with the re-write. Thanks for sharing the issue and the fix!

**Comment by olaulau** [comment]
same with apache-fpm. funciton will be available back again in FPM from PHP 7.3 :-/
